File: 21451198卢超/project4/Note/Note/AddNoteViewController.m
//
//  AddNoteViewController.m
//  Note
//
//  Created by jiaoshoujie on 14-11-21.
//  Copyright (c) 2014年 ___FULLUSERNAME___. All rights reserved.
//

#import "AddNoteViewController.h"
#import "sqlite3.h"
#define kDatabaseName @"database.sqlite3"

@interface AddNoteViewController ()

@end

@implementation AddNoteViewController

@synthesize databaseFilePath;
@synthesize rightButon;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    //获取数据库文件路径
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *documentsDirectory = [paths objectAtIndex:0];
    self.databaseFilePath = [documentsDirectory stringByAppendingPathComponent:kDatabaseName];
}

//关闭键盘
- (IBAction)backgroundTap:(id)sender {
    [_textView resignFirstResponder];
}

-(void)viewDidAppear:(BOOL)animated{
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)addNoteAction:(id)sender {
    //UIAlertView *alter = [[UIAlertView alloc] initWithTitle:@"提示" message:@"你点击了导航栏右按钮" delegate:self  cancelButtonTitle:@"确定" otherButtonTitles:nil, nil];
    //[alter show];
    NSString *title = [[NSString alloc] initWithCString:"完成" encoding:NSUTF8StringEncoding];
    NSLog(@"title:%@,rightButton:%@",title,rightButon.title);
    NSLog(@"textView:%@",_textView.text);
    if ([title isEqualToString:rightButon.title]&&![[[NSString alloc]initWithCString:"" encoding:NSUTF8StringEncoding]isEqualToString:_textView.text] ) {
        [self insertIntoDatabase];
        [rightButon setTitle:@"添加"];
        _textView.text = nil;
    }else{
        [rightButon setTitle:@"完成"];
    }
}

-(void)insertIntoDatabase{
    //打开数据库
    sqlite3 *database;
    if (sqlite3_open([self.databaseFilePath UTF8String], &database)!=SQLITE_OK) {
        sqlite3_close(database);
        NSAssert(0, @"打开数据库失败！");
    }
    int last = [self lastTagofNotes];
    //使用约束变量插入数据
    char *update = "INSERT OR REPLACE INTO NOTES (TAG, DETAILS, THEDATETIME) VALUES (?,?,?);";
    sqlite3_stmt *stmt;
    if (sqlite3_prepare_v2(database, update, -1, &stmt, nil) == SQLITE_OK) {
        sqlite3_bind_int(stmt, 1, last+1);
        sqlite3_bind_text(stmt, 2, [_textView.text UTF8String], -1, NULL);
        NSDate* date = [NSDate date];
        NSDateFormatter* formatter = [[NSDateFormatter alloc] init];
        [formatter setDateFormat:@"yyyy-MM-dd HH:MM:SS"];
        NSString* str = [formatter stringFromDate:date];
        NSLog(@"现在时间是：%@",str);
        sqlite3_bind_text(stmt, 3, [str UTF8String], -1, NULL);
    }
    char *errorMsg = NULL;
    if (sqlite3_step(stmt) != SQLITE_DONE) {
        NSAssert(0, @"更新数据库表FIELDS出错：%s",errorMsg);
    }
    sqlite3_finalize(stmt);
    //关闭数据库
    sqlite3_close(database);
}

-(int)lastTagofNotes{
    //打开数据库
    sqlite3 *database;
    if (sqlite3_open([self.databaseFilePath UTF8String], &database)!=SQLITE_OK) {
        sqlite3_close(database);
        NSAssert(0, @"打开数据库失败！");
    }
    int lastTag = 0;
    char *update = "SELECT MAX(TAG) FROM NOTES";
    sqlite3_stmt *stmt;
    if (sqlite3_prepare_v2(database, update, -1, &stmt, nil) == SQLITE_OK){
        while (sqlite3_step(stmt)==SQLITE_ROW){
            lastTag = sqlite3_column_int(stmt, 0);
        }
        sqlite3_finalize(stmt);
    }
    sqlite3_close(database);
    
    return lastTag;
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: yxb144325165潘云童/第四次作业/MyNote/MyNote/AddNoteViewController.swift
//
//  AddNoteViewController.swift
//  MyNote
//
//  Created by Joker on 14/11/21.
//  Copyright (c) 2014年 Joker. All rights reserved.
//

import UIKit
import CoreData

class AddNoteViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate, UITextViewDelegate {

    @IBOutlet weak var timeLabel: UILabel!
    @IBOutlet weak var image: UIImageView!
    @IBOutlet weak var inputTextView: UITextView!
    
    var num = ["0", "1", "2"]
    var imageName = ""
    
    // 打开图库
    @IBAction func getImage(sender: UIButton) {
        var c = UIImagePickerController()
        c.sourceType = UIImagePickerControllerSourceType.PhotoLibrary
        c.delegate = self
        
        self.presentViewController(c, animated: true, completion: nil)
    }
    
    // 取消笔记
    @IBAction func cancel(sender: UIBarButtonItem) {
        self.navigationController?.popViewControllerAnimated(true)
    }
    
    // 保存笔记
    @IBAction func save(sender: UIBarButtonItem) {
//        println(self.navigationController?.viewControllers)
        
        // 插入一条新的笔记
        let myNote = self.navigationController?.viewControllers.first as MyNoteTableViewController
        let context = myNote.fetchedResultsController.managedObjectContext
        let newNote = NSEntityDescription.insertNewObjectForEntityForName("Note", inManagedObjectContext: context) as Note
        newNote.time = timeLabel.text!
        newNote.text = inputTextView.text!
        if (image.image != nil) {
            saveImage(image.image!)
            newNote.image = imageName
        } else {
            newNote.image = ""
        }
        newNote.managedObjectContext?.save(nil)
        
        self.navigationController?.popViewControllerAnimated(false)
    }
    
    func textViewDidBeginEditing(textView: UITextView) {
        if (inputTextView.text == "点击编辑") {
            inputTextView.text = ""
            inputTextView.textColor = UIColor.blackColor()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.navigationController?.navigationBarHidden = true
        inputTextView.delegate = self
        
        //获取当前时间
        let now = NSDate()
        var df = NSDateFormatter()
        df.setLocalizedDateFormatFromTemplate("yyyy-MM-dd HH:mm")
        let time = df.stringFromDate(now)
        timeLabel.text = time
        
        df.setLocalizedDateFormatFromTemplate("yyyy-MM-dd HH:mm:ss")
        imageName = formatTime(df.stringFromDate(now)) + ".png"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    //将选择的图片保存到该笔记中
    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [NSObject : AnyObject]) {
        
        image.image = info[UIImagePickerControllerOriginalImage] as? UIImage

        picker.dismissViewControllerAnimated(true, completion: nil)
    }
    
    func saveImage(noteImage: UIImage) {
        let paths = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)
        let filePath = paths.first?.stringByAppendingPathComponent(imageName) as String!
        imageName = filePath
        UIImagePNGRepresentation(image.image).writeToFile(filePath, atomically: true)
    }
    
    func formatTime(time: String) ->String {
        var format = ""
        for char in time {
            if (char != "/" && char != ":" && char != " " && char != ",") {
                format += String(char)
            }
        }
        return format
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
